packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
+/// any size specified in the child of an [Expanded] will be ignored,
+/// as the child will be forced to be the size given by its tight
+/// constraints.
+///
+/// [FlexFit.loose] is the default fit for a [Flexible]. It will allow
+/// the child to be the size the child's size parameters state it
+/// should be, as long as that size is not outside the range defined by
+/// the minimum and maximum constraints given to the [Flexible].
+///
+/// [MainAxisSize] : An enum that determines if the [Flex] will be
+/// forced to be as large as its parent will allow.
+///
+/// [MainAxisSize.max] will force the [Flex] to be as large as the parent
+/// will allow in the main axis. _This does not force the children of the
+/// [Flex] to be larger._ This is the default.
+///
+/// [MainAxisSize.min] : Allows the [Flex] to be whatever size it needs
+/// to be in order to accommodate its children, without forcing the
+/// [Flex] to be any larger than the total size of all its children.
+///
+/// [MainAxisAlignment] : How to align, center, or apply spacing
+/// between the children in the direction of the main axis.
+///
+/// [CrossAxisAlignment] : How to align, or stretch, the children in
+/// the direction of the cross axis.
+///
+/// The multi-child layout algorithm proceeds in six steps:
+///
+/// 1. Lay out each child that is not a [Flexible] or an [Expanded],
+///    without setting minimum or maximum constraints (sizes) in the
+///    main axis. So, anything that is not a [Flexible] or an [Expanded]
+///    is going to get laid out but without setting a specific size in the
+///    main axis, yet.
+///
+///    If [crossAxisAlignment] is set to anything other than
+///    [CrossAxisAlignment.stretch], apply any constraints in the
+///    _cross axis_ direction that may have been passed down by the
+///    parent of the [Flex] without forcing the child to be any larger than
+///    it's stated size in that direction. This is called using _"loose
+///    constraints"_ (in this case, in the cross axis).
+///    Note that these are constraints coming from the _parent_ of the
+///    [Flex], not the [CrossAxisAlignment] of the [Flex].
+///
+///    However, if the [CrossAxisAlignment] of the [Flex] has been
+///    set to [CrossAxisAlignment.stretch] then force all the children of the
+///    [Flex] to be as large as possible in the cross axis direction. This is
+///    achieved by using _"tight constraints"_. In such a case, the maximum
+///    cross axis size of the children of the [Flex] will be forced to be the
+///    same as the maximum constraint (size) _in that direction_ (height or
+///    width) that was passed down by the parent of the [Flex].
+///
+///    If the parent of the [Flex] has not set any size limit in the cross
+///    axis, and the child is trying to be the max possible size in that
+///    axis, then the error you will see is likely _"BoxConstraints forces an
+///    infinite (height or width)"_.
```

Comment:
Maybe add a sentence on how to fix it?  Something like "To fix this, a specific size must be specified, or a maximum constraint introduced."

Reply:
Handling this offline in the IDE.